pyomo/contrib/piecewise/transform/disagreggated_logarithmic.py

```diff
@@ -0,0 +1,178 @@
+from pyomo.contrib.piecewise.transform.piecewise_to_gdp_transformation import (
+    PiecewiseLinearTransformationBase,
+)
+from pyomo.core import Constraint, Binary, Var, RangeSet
+from pyomo.core.base import TransformationFactory
+from pyomo.common.errors import DeveloperError
+from math import ceil, log2
+
+
+@TransformationFactory.register(
+    "contrib.piecewise.disaggregated_logarithmic",
+    doc="""
+    Represent a piecewise linear function "logarithmically" by using a MIP with
+    log_2(|P|) binary decision variables. This is a direct-to-MIP transformation; 
+    GDP is not used. This method of logarithmically formulating the piecewise 
+    linear function imposes no restrictions on the family of polytopes, but we 
+    assume we have simplces in this code. This method is due to Vielma et al., 2010.
+    """,
+)
+class DisaggregatedLogarithmicInnerMIPTransformation(PiecewiseLinearTransformationBase):
+    """
+    Represent a piecewise linear function "logarithmically" by using a MIP with
+    log_2(|P|) binary decision variables. This is a direct-to-MIP transformation;
+    GDP is not used. This method of logarithmically formulating the piecewise
+    linear function imposes no restrictions on the family of polytopes, but we
+    assume we have simplces in this code. This method is due to Vielma et al., 2010.
```

Comment:
I would add the full reference somewhere

Reply:
Added the full reference